### DISCUSSION TITLE
How could libp2p help this project?

### DISCUSSION THREAD

**Discussion by ghost** [libp2p/rust-libp2p#2740] [Q&A] [answered]
How could libp2p help a multi-protocol p2p file-sharing program similar to [MLDonkey](https://github.com/ygrek/mldonkey), [Shareaza](https://github.com/ivan386/Shareaza) or [Envy](https://github.com/GetEnvy/Envy). The project adapts spacedrive's codebase that already uses libp2p.

**Comment by mxinden** [comment, marked as answer]
I am not sure I see the question. Can you be more concrete?

**Reply by ghost** [reply]
Would it make it easier to implement p2p networks like ed2k, kademlia, dht, bittorrent, direct connect, soulseek, gnutella..?

**Reply by elenaf9** [reply]
Yes. Libp2p is a modular system of protocols, specifications and libraries that enable the development of p2p network applications.
Maybe the docs can help you to explore if / how libp2p can help your project https://docs.libp2p.io/introduction/what-is-libp2p/ :)
